mirdata/track.py

```diff
@@ -33,11 +33,78 @@ def __repr__(self):
             if val.__doc__ is None:
                 raise ValueError("{} has no documentation".format(prop))
 
-            val_type_str = val.__doc__.split(':')[0]
+            val_type_str = val.__doc__.split(":")[0]
             repr_str += "  {}: {},\n".format(prop, val_type_str)
 
         repr_str += ")"
         return repr_str
 
     def to_jams(self):
         raise NotImplementedError
+
+
+class MultiTrack(Track):
+    """MultiTrack class.
+
+    A multitrack class is a collection of track objects and their associated audio
+    that can be mixed together.
+    A multitrack is iteslf a Track, and can have its own associated audio (such as
+    a mastered mix), its own metadata and its own annotations.
+
+    Attributes:
+        tracks (dict): {track_id: Track}
+        track_audio_attribute (str): the name of the attribute of Track which 
+            returns the audio to be mixed
+    """
+
+    def __init__(self, tracks, track_audio_attribute):
+        """Inits MultiTrack with tracks and audio attribute"""
+        self.tracks = tracks
+        self.track_audio_attribute = track_audio_attribute
+
+    def get_target(self, track_keys, weights=None):
+        """Create a linear mixture given a subset of tracks.
+
+        Args:
+            track_keys (list): list of track keys to mix together
+            weights (list or None): list of positive scalars to be used in the average
+        
+        Returns:
+            target (np.ndarray): mixture audio with shape (n_samples, n_channels)
+        """
+        signals = [
+            getattr(self.tracks[k], self.track_audio_attribute)() for k in track_keys
+        ]
+        return np.average(signals, axis=0, weights=weights)
```

Comment:
sometimes we don't want to average the signals and take the sum (multichannel)
in this case maybe we can add a parameter average=True 
`if average:
    return np.average(signals, axis=0, weights=weights)
else:
    return np.sum(signals, axis=0, weights=weights)`

Reply:
also this would give an error if the signals have different lengths

Reply:
OK, will update!

+1 for catching arrays with different lengths